### ISSUE TITLE
URL rewriting does not handle adapter being behind HTTPS load balancer

### ISSUE THREAD

**Issue by no-man-is-an-island** [nhsconnect/integration-adaptor-gpc-consumer#60]
Affects version: 0.2.6

When rewriting the URLs, the protocol is set to plain HTTP:

```
2021-09-16 10:38:16.820 Level=INFO Logger=u.n.a.g.c.f.UrlsInResponseBodyRewriteFunction Thread="reactor-http-epoll-4" Message="Replacing all occurrences of 'https://Test1-N82103.evahealth.thirdparty.nhs.uk' in the response body with 'http://infrastructure-david-mhs-gpc.infra.newhealth.care'"
```

In our infrastructure the GPCC adapter is deployed behind an ALB, so the `x-forwarded-proto` header would need to be checked to establish the correct protocol to use for the rewriting

**Comment by no-man-is-an-island** [comment]
Issue remains on 0.2.7 - to clarify, the header is on the initial request to the gpcc adapter, not on the response from SSP/gpc provider

**Comment by no-man-is-an-island** [comment]
Resolved in 0.2.8
